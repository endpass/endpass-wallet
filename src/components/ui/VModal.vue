<template lang="html">
  <div class="modal is-active">
    <div
      class="modal-background"
      @click="close"
    />
    <div class="modal-logo" />
    <div class="modal-card">
      <div class="modal-card-head">
        <h3 class="modal-card-title">
          <slot name="header" />
        </h3>
        <button
          class="delete is-large"
          @click="close" />
      </div>
      <div class="modal-card-body">
        <slot />
      </div>
      <div class="modal-card-foot">
        <slot name="footer" />
      </div>
    </div>
  </div>
</template>

<script>
// Generic modal component to be used with slots
export default {
  name: 'VModal',
  inject: ['$validator'],
  methods: {
    close() {
      this.$emit('close');
    },
  },
};
</script>

<style lang="scss">
.modal {
  .modal-card {
    .modal-card-head {
      background-color: $purple;
      border-bottom: none;
    }
    .modal-card-title {
      color: $white;
    }
    .modal-card-foot {
      background-color: $white;
      padding: 2rem 1rem;
      text-align: center;
      //border-radius: 0;
      // Full width buttons
      .buttons {
        flex: 1;

        .button {
          &:not(:last-child) {
            margin-right: 0;
          }
        }
      }
    }
  }
}

.modal.is-dark {
  .modal-background {
    background-color: $purple;
  }

  .modal-card {
    .modal-card-head {
      background-color: white;
      border-bottom: 1px solid $purple;
      color: $purple;
    }

    .modal-card-title {
      color: $purple;
    }
  }

  .delete {
    background-color: $purple;
    &:hover,
    &.is-hoevered {
      background-color: lighten($purple, 5%);
      transition: background-color 0.2s ease-in;
    }
  }

  .modal-logo {
    position: absolute;
    top: 1rem;
    left: 1rem;
    width: 240px;
    height: 80px;
    background-size: cover;
    background-image: url('/static/logo-dark.png');
  }
  @media screen and (max-width: 1023px) {
    .modal-logo {
      width: 144px;
      height: 48px;
    }
  }
}

.modal.is-naked {
  .modal-card {
    .modal-card-head,
    .modal-card-foot,
    .modal-card-body {
      background-color: transparent;
      color: $white;
    }
    .modal-card-title {
      color: $white;
    }

    .modal-card-body {
      .label {
        color: $white;
      }
    }
  }
  .delete {
    background-color: transparent;
  }
}
</style>
