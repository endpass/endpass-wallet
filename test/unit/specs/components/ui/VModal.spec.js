import { shallow, createLocalVue } from '@vue/test-utils';
import VeeValidate from 'vee-validate';

import VModal from '@/components/ui/VModal';

const localVue = createLocalVue();

localVue.use(VeeValidate);

describe('VModal page', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(VModal, {
      localVue,
      provide: () => ({
        $validator: new VeeValidate.Validator(),
      }),
    });
  });

  describe('render', () => {
    it('should be a Vue component', () => {
      expect(wrapper.name()).toBe('v-modal');
      expect(wrapper.isVueInstance()).toBeTruthy();
    });

    it('should render initial state of the component', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
