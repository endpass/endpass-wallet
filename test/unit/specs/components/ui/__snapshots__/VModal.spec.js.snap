// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VModal page render should render initial state of the component 1`] = `
<div
  class="modal is-active"
>
  <div
    class="modal-background"
  />
   
  <div
    class="modal-logo"
  />
   
  <div
    class="modal-card"
  >
    <div
      class="modal-card-head"
    >
      <h3
        class="modal-card-title"
      />
       
      <button
        class="delete is-large"
      />
    </div>
     
    <div
      class="modal-card-body"
    />
     
    <div
      class="modal-card-foot"
    />
  </div>
</div>
`;
